packages/hardhat-chai-matchers/test/reverted/reverted.ts

```diff
@@ -0,0 +1,362 @@
+          matchers,
+          method: "revertsWith",
+          args: ["some reason"],
+          successfulAssert: (x) => expect(x).to.be.reverted,
+        });
+      });
+
+      it("failed asserts", async function () {
+        await runFailedAsserts({
+          matchers,
+          method: "revertsWith",
+          args: ["some reason"],
+          failedAssert: (x) => expect(x).not.to.be.reverted,
+          failedAssertReason:
+            "Expected transaction NOT to be reverted, but it reverted with reason 'some reason'",
+        });
+      });
+    });
+
+    describe("calling a method that reverts with a panic code", function () {
+      it("successful asserts", async function () {
+        await runSuccessfulAsserts({
+          matchers,
+          method: "panicAssert",
+          args: [],
+          successfulAssert: (x) => expect(x).to.be.reverted,
+        });
+      });
+
+      it("failed asserts", async function () {
+        await runFailedAsserts({
+          matchers,
+          method: "panicAssert",
+          args: [],
+          failedAssert: (x) => expect(x).not.to.be.reverted,
+          failedAssertReason:
+            "Expected transaction NOT to be reverted, but it reverted with panic code 0x01 (Assertion error)",
+        });
+      });
+    });
+
+    describe("calling a method that reverts with a custom error", function () {
+      it("successful asserts", async function () {
+        await runSuccessfulAsserts({
+          matchers,
+          method: "revertWithSomeCustomError",
+          args: [],
+          successfulAssert: (x) => expect(x).to.be.reverted,
+        });
+      });
+
+      it("failed asserts", async function () {
+        await runFailedAsserts({
+          matchers,
+          method: "revertWithSomeCustomError",
+          args: [],
+          failedAssert: (x) => expect(x).not.to.be.reverted,
+          failedAssertReason: "Expected transaction NOT to be reverted",
+        });
+      });
+    });
+
+    describe("invalid rejection values", function () {
+      it("non-errors", async function () {
+        await expectAssertionError(
+          expect(Promise.reject({})).to.be.reverted,
+          "Expected an Error object"
+        );
+      });
+
+      it("errors that are not related to a reverted transaction", async function () {
+        // use an address that almost surely doesn't have balance
+        const randomPrivateKey =
+          "0xc5c587cc6e48e9692aee0bf07474118e6d830c11905f7ec7ff32c09c99eba5f9";
+        const signer = new this.hre.ethers.Wallet(
+          randomPrivateKey,
+          this.hre.ethers.provider
+        );
+
+        // this transaction will fail because of lack of funds, not because of a
+        // revert
+        await expect(
+          expect(
+            matchers.connect(signer).revertsWithoutReasonString({
+              gasLimit: 1_000_000,
+            })
+          ).to.not.be.reverted
+        ).to.be.eventually.rejectedWith(
```

Comment:
Perhaps add `ProviderError` as the first argument to this `rejectedWith()`?

Reply:
Makes sense, done in 861b61f89818ae4b8a762f0cf047c8696928d215